File: Data Structures & Algorithms/max-water-container/submission-0.java
class Solution {
    public int maxArea(int[] heights) {
        int a = 0;
        int b = heights.length-1;
        int maxS = 0;
        while(a<b){
            int min = (heights[a]<heights[b])?a:b;
            int s = (b-a)*heights[min];
            maxS = Math.max(s,maxS);
            if(min==a)a++;
            else b--;
        }
        return maxS;
    }
}

File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-0.kt
class Solution {
    fun findMin(nums: IntArray): Int {
        var l = 0
        var m = nums.size/2
        var r = nums.size-1
        while(l!=m&&r!=m){
            if(nums[r]<nums[m]){
                l=m
                m=l+(r-l)/2
            }else
            if(nums[l]>nums[m]){
                r=m
                m=l+(r-l)/2
            }else{
                break
            }
        }
        return Math.min(nums[l],nums[r])
    }
}